iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationCheckUtils.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.common.checks;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PrivilegeEscalationCheckUtils {
+
+  private PrivilegeEscalationCheckUtils() {
+    // utils class
+  }
+
+  private static final Set<PrivilegeEscalationVector> ESCALATION_VECTORS = Set.of(
+    new PrivilegeEscalationVector("iam:CreatePolicyVersion"),
+    new PrivilegeEscalationVector("iam:SetDefaultPolicyVersion"),
+    new PrivilegeEscalationVector("iam:CreateAccessKey"),
+    new PrivilegeEscalationVector("iam:CreateLoginProfile"),
+    new PrivilegeEscalationVector("iam:UpdateLoginProfile"),
+    new PrivilegeEscalationVector("iam:AttachUserPolicy"),
+    new PrivilegeEscalationVector("iam:AttachGroupPolicy"),
+    new PrivilegeEscalationVector("iam:AttachRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:PutUserPolicy"),
+    new PrivilegeEscalationVector("iam:PutGroupPolicy"),
+    new PrivilegeEscalationVector("iam:PutRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:AddUserToGroup"),
+    new PrivilegeEscalationVector("iam:UpdateAssumeRolePolicy", "sts:AssumeRole"),
+    new PrivilegeEscalationVector("iam:PassRole", "ec2:RunInstances"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:InvokeFunction"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:AddPermission"),
+    new PrivilegeEscalationVector("iam:PassRole", "lambda:CreateFunction", "lambda:CreateEventSourceMapping"),
+    new PrivilegeEscalationVector("iam:PassRole", "cloudformation:CreateStack"),
+    new PrivilegeEscalationVector("iam:PassRole", "datapipeline:CreatePipeline", "datapipeline:PutPipelineDefinition"),
+    new PrivilegeEscalationVector("iam:PassRole", "glue:CreateDevEndpoint"),
+    new PrivilegeEscalationVector("glue:UpdateDevEndpoint"),
+    new PrivilegeEscalationVector("lambda:UpdateFunctionCode")
+  );
+
+  public static boolean escalationVectorApply(Stream<String> actionPermissions) {
```

Comment:
I think the name of this method could better show that it returns a boolean.
Maybe something like `isSupersetOfAnEscalationVector`?